stock_account_move_multi_reset_to_draft/models/account_move.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def button_draft(self):
+        """If it is a purchase invoice, we will delete the SVLS linked to it."""
+        res = super().button_draft()
+        for item in self.filtered(
+            lambda x: x.is_inbound
+            and any(line.stock_valuation_layer_ids for line in x.sudo().line_ids)
+        ):
+            item.sudo().line_ids.stock_valuation_layer_ids.unlink()
```

Comment:
Perhaps a better option would be to create a copy of stock_valuation_layer_ids but with opposite sign.
This will ensure that if someone print a stock valuation report on a certain date, that report will not be affected by later invoice changes.
Otherwise users may say "How come I have this report in pdf with one value and if I pull it now it comes out with a different value?